lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1312,10 +1312,15 @@ sub _failure_reason {
     my @failed_modules;
     my $modules = $self->modules;
     while (my $m = $modules->next) {
-        next if ($m->result eq INCOMPLETE);
-        push(@failed_modules, $m->name . ':' . $m->result);
+        if ($m->result eq FAILED) {
+            # in case we don't see an important module, we return all modules
+            push(@failed_modules, $m->name . ':' . $m->result);
+        }
+        if ($m->result eq FAILED && $m->fatal) {
+            return $m->name;
+        }
```

Comment:
do we need this special handling or is it fine what @aaannz proposed (see above, adding all modules)